### PR TITLE
auth: remote workspaces use the global state, showing incorrect profile info. 

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
In Title. 
## Solution
We check if we are on remote by using `vscode.env.remoteName`. Then we use workspace state instead of global state when on a non-CodeCatalyst remote instance. 

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by JadenSimon — APPROVED** [review]
Works correctly on CodeCatalyst.

**Comment by justinmk3** [comment]
Which commits are relevant? Github is showing change that are already on master, for some reason. Try squash-merge and rebase.

**Comment by Hweinstock** [comment]
Believe I fixed it by merging in master. Will make sure to squash and merge when doing final merge.

File: src/auth/auth.ts [hunk 1]
```diff
@@ -743,11 +743,21 @@ export class Auth implements AuthService, ConnectionManager {
 
     static #instance: Auth | undefined
     public static get instance() {
-        const devEnvId = getCodeCatalystDevEnvId()
-        const memento =
-            devEnvId !== undefined ? partition(globals.context.globalState, devEnvId) : globals.context.globalState
+        return (this.#instance ??= new Auth(new ProfileStore(getMemento())))
 
-        return (this.#instance ??= new Auth(new ProfileStore(memento)))
+        function getMemento() {
```
Comment (justinmk3) [inline src/auth/auth.ts:748]: looks strange to define a function-scoped function after `return`

**Comment by justinmk3** [comment]
Is this a correct Problem/Solution framing:

> Problem:
> When connected to remote EC2 instance, auth info is wrong.
>
> Solution:
> In a non-CodeCatalyst remote context, get auth info from "workspace state" instead of "global state".

**Pull request merged**